content_scripts/vimium_frontend.coffee

```diff
@@ -96,6 +96,30 @@ settings =
 #
 frameId = Math.floor(Math.random()*999999999)
 
+# If an input grabs the focus before the user has interacted with the page, then grab it back.
+class GrabBackFocus extends Mode
+  constructor: ->
+    super
+      name: "grab-back-focus"
+      keydown: => @alwaysContinueBubbling => @exit()
+
+    @push
+      _name: "grab-back-focus-mousedown"
+      mousedown: => @alwaysContinueBubbling => @exit()
+
+    chrome.storage.sync.get "grabBackfocus", (items) =>
```

Comment:
> If those are things we need the settings library to do, we should do them there

I agree.  I was thinking about whether to do it in this PR, or wait.  I'd also like to have a `cachedSetting` object to handle the synchronous case cleanly and consistently.